areas/quota/src/AzureMcp.Quota/Services/Util/Usage/ContainerAppUsageChecker.cs

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Azure.Core;
+using Azure.ResourceManager.AppContainers;
+using Azure.ResourceManager.AppContainers.Models;
+using Microsoft.Extensions.Logging;
+
+namespace AzureMcp.Quota.Services.Util;
+
+public class ContainerAppUsageChecker(TokenCredential credential, string subscriptionId, ILogger<ContainerAppUsageChecker> logger) : AzureUsageChecker(credential, subscriptionId, logger)
+{
+    public override async Task<List<UsageInfo>> GetUsageForLocationAsync(string location)
+    {
+        try
+        {
+            var subscription = ResourceClient.GetSubscriptionResource(new ResourceIdentifier($"/subscriptions/{SubscriptionId}"));
+            var usages = subscription.GetUsagesAsync(location);
+            var result = new List<UsageInfo>();
+
+            await foreach (var item in usages)
+            {
+                result.Add(new UsageInfo(
+                    Name: item.Name?.Value ?? string.Empty,
+                    Limit: (int)(item.Limit),
+                    Used: (int)(item.CurrentValue),
+                    Unit: item.Unit.ToString()
+                ));
+            }
+
+            return result;
+        }
+        catch (Exception error)
+        {
+            throw new InvalidOperationException("Failed to fetch Container Apps quotas. Please check your subscription permissions and service availability.", error);
+        }
+    }
```

Comment:
All of these usage checkers are exactly the same - why do we need multiple implementations?  They are calling the same SDKs, returning the same data elements.